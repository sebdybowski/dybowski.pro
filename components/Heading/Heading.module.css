.Heading {
    position: relative;
    margin: 2rem 0;
}

h1.Heading {
    font-size: 6rem;
}

h2.Heading {
    font-size: 3rem;
}

.Heading--serif {
    font-family: 'Times New Roman', Times, serif;
}

.Heading--brand {
    color: var(--primary-color);
}

.Heading--section::after {
   content: '';
   width: 2rem;
   height: 0.75rem;
   background-color: var(--text-color);
   position: absolute;
   border-radius: 0.5rem;
   bottom: -2rem;
   left: calc(50% - 1rem);
}