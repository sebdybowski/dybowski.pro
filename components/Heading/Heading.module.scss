@use 'styles/variables.scss' as *;

.Heading {
  position: relative;
  margin: 1rem 0;
}

h1.Heading {
  font-size: 3rem;
}

h2.Heading {
  font-size: 2rem;
}

.Heading--serif {
  font-family: $display-font;
}

.Heading--brand {
  color: $text-color-dark;

  @media screen and (max-width: ($breakpoint-md - 1)) {
    background-color: $text-color-light;
    padding: 0.25rem 0.5rem;
  }
}

.Heading--section::after {
  content: '';
  width: 0.5rem;
  height: 0.5rem;
  background-color: $text-color-dark;
  position: absolute;
  border-radius: 0.5rem;
  bottom: -1.8rem;
  left: calc(50% - 0.25rem);
}