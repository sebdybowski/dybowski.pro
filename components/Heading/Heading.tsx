import { ReactNode, createElement } from 'react';
import clsx from 'clsx';

import styles from './Heading.module.css';

type HeadingProps = {
  children: ReactNode,
  serif?: boolean,
  brand?: boolean,
  level?: 1 | 2 | 3 | 4 | 5 | 6,
  section?: boolean
};

const Heading = ({
  level = 1,
  children,
  serif = false,
  brand = false,
  section = false,
}: HeadingProps) => createElement(
  `h${level}`,
  {
    className: clsx(
      styles.Heading,
      serif && styles['Heading--serif'],
      brand && styles['Heading--brand'],
      section && styles['Heading--section'],
    ),
    ['data-testid']: 'Heading',
  },
  children,
);

export default Heading;
