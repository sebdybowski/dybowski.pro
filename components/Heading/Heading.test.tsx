import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Heading from './Heading';

describe('<Heading />', () => {
  test('it should mount', () => {
    render(<Heading />);
    
    const heading = screen.getByTestId('Heading');

    expect(heading).toBeInTheDocument();
  });
});