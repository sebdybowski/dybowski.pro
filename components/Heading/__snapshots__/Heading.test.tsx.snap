// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> it should mount 1`] = `
<DocumentFragment>
  <h1
    class="Heading"
    data-testid="Heading"
  >
    Heading!
  </h1>
</DocumentFragment>
`;

exports[`<Heading /> it should mount with optional props 1`] = `
<DocumentFragment>
  <h1
    class="Heading Heading--serif Heading--brand Heading--section"
    data-testid="Heading"
  >
    Heading!
  </h1>
</DocumentFragment>
`;
